components/cascader/index.tsx

```diff
@@ -317,10 +320,30 @@ export default class Cascader extends React.Component<CascaderProps, CascaderSta
       filter = defaultFilterOption,
       render = defaultRenderFilteredOption,
       sort = defaultSortFilteredOption,
+      limit = defaultLimit,
     } = showSearch as ShowSearchType;
     const { flattenOptions = [], inputValue } = this.state;
-    const filtered = flattenOptions.filter((path) => filter(this.state.inputValue, path, names))
-      .sort((a, b) => sort(a, b, inputValue, names));
+
+    // Limit the filter if needed
+    let filtered: Array<CascaderOptionType[]>;
+    if (limit > 0) {
+      filtered = [];
+      let matchCount = 0;
+
+      // Perf optimization to filter items only below the limit
+      flattenOptions.some((path) => {
+        const match = filter(this.state.inputValue, path, names);
+        if (match) {
+          filtered.push(path);
+          matchCount += 1;
+        }
+        return matchCount >= limit;
+      });
+    } else {
+      filtered = flattenOptions.filter((path) => filter(this.state.inputValue, path, names));
```

Comment:
As `limit` is backed by a default value, a user can only specify
```
<Cascader showSearch={{ filter, limit: -1 }} />
```
to go into this branch.

I suggest just cap the `limit` to be non-negative and remove this branch.